libraries/AP_NavEKF3/AP_NavEKF3_Outputs.cpp

```diff
@@ -119,7 +119,7 @@ bool NavEKF3_core::getRangeBeaconDebug(uint8_t &ID, float &rng, float &innov, fl
 bool NavEKF3_core::getHeightControlLimit(float &height) const
 {
     // only ask for limiting if we are doing optical flow navigation
-    if (frontend->_fusionModeGPS == 3 && (PV_AidingMode == AID_RELATIVE) && flowDataValid) {
+    if (frontend->sources.useVelXYSource(AP_NavEKF_Source::SourceXY::OPTFLOW) && (PV_AidingMode == AID_RELATIVE) && flowDataValid) {
```

Comment:
The previous configuration didn't allow the user to specify what non-GPS source they were using but it's clear from the comments that this height limiting should only apply while optical flow is being used.  The new configuration does allow us to be more specific so the new code makes use of this.

re "are" vs "could", I'm not sure there is any change.  We could replace the "useVelXYSource(OPTFLOW)" with "getVelXYSource() == OPTFLOW" but I don't think it matters because of the additional "PV_AidingMode == AID_RELATIVE" check.

I think we should limit the height if we are trying to fuse optical flow and the GPS, Vicon or Beacon sensors aren't  working well enough to give the vehicle an AID_ABSOLUTE yet (or they haven't been configured to be used).

I think there are situations where this logic (both before and after my change) could go wrong:

- PV_AidingMode == Absolute (i.e. we had a good GPS vicon or beacon position at some point) but have since lost those sensors (i.e. they have been disabled or their data is too bad to be fused) and we are only using optical flow.  In this case no height limiting will be applied but it probably should be.  I don't think the existing code would have applied limits either though.

- ExtlNav is *not* providing position but it is providing speed estimates and/or position deltas and optical flow is also being applied.  In this case height limiting will be applied when it probably shouldn't be assuming the ExtNav system can work at any altitude (like the T265).

Reply:
Decided not to act on this problem; pre-existing.

Reply:
we discussed this and I think PeterB came around to agreeing with me so marking as resolved.